From 266efbc76916f6d60f7d36db0fcb6cd83d668f37 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Sun, 05 Jun 2022 18:33:41 +0300
Subject: [PATCH] Revert "[SW-77251] gaudi3: add start image api"

Revert rom recovery implementation for secured firmware update via
inband and OOB.

This reverts commit 6c0846b188d83c933f4f497b8d3d65e3a94b8d97.

Change-Id: Ib8978d4519026e824828b749a2b4675a915aafc2
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 35c4e5e..6b054a5 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -865,20 +865,6 @@
 	return false;
 }
 
-void arcboot_start_image(uint64_t startaddr)
-{
-	void (*image_entry)(void);
-	uint32_t arc_addr;
-
-	if (!startaddr)
-		wfe();
-
-	arc_addr = (uint32_t)arcboot_rom_get_arc_addr(startaddr);
-
-	image_entry = (void *)((uint32_t) arc_addr);
-	image_entry();
-}
-
 void arcboot_rom_in_recovery(void)
 {
 	uint8_t override = 1;
diff --git a/include/arcboot.h b/include/arcboot.h
index e9f1db2..de29ef8 100644
--- a/include/arcboot.h
+++ b/include/arcboot.h
@@ -270,7 +270,6 @@
 int arcboot_rom_mem_init(uint64_t addr, uint32_t size);
 void arcboot_rom_verify(void);
 bool arcboot_is_valid_address_range(uint64_t addr, int len);
-void arcboot_start_image(uint64_t startaddr);
 uint32_t arcboot_error_read_and_clr(uint32_t *major, uint32_t *minor);
 bool arcboot_error_inform(uint32_t major, uint32_t minor);
 bool arcboot_error_is_pending(void);
